docs/raw/serving-community-history.md

```diff
@@ -84,200 +89,282 @@ Furthermore, it assumes that:
+The `offset` field MUST contain the position at which the message history archive starts in the byte string of the total message archive data. This MUST be the sum of the length of all previously created message archives in bytes (see [Creating message archive torrents](#creating-message-archive-torrents)).
 
-A `WakuMessageArchiveIndex` is a map where the key is the KECCAK-256 hash of the magnet link derived from the previously created archive and the value an instance of `WakuMessageArchiveMetadata`, which is also derived from the previously created message archive.
+```
+syntax = "proto3"
 
-### WakuMessageArchiveMetadata
+message WakuMessageArchiveIndexMetadata {
+  uint8 version = 1
+  WakuMessageArchiveMetadata metadata = 2
+  uint64 offset = 3
+  uint64 num_pieces = 4
+}
 
-The `from` field SHOULD contain a timestamp of the time range's lower bound.
+message WakuMessageArchiveIndex {
+  map<string, WakuMessageArchiveIndexMetadata> archives = 1
+}
+```
 
-The `to` field SHOULD contain a timestamp of the time range's the higher bound.
+The community owner node MUST update the `WakuMessageArchiveIndex` every time it creates one or more `WakuMessageArchive`s and bundle it into a new torrent (**TODO: see section**).
+For every created `WakuMessageArchive`, there MUST be a `WakuMessageArchiveIndexMetadata` entry in the `archives` field `WakuMessageArchiveIndex`.
 
-The `contentTopic` field MUST contain the same `contentTopic` that the archive's `messages` have.
+## Creating message archive torrents
 
-The `magnet_uri_hash` field MUST contain the KECCAK-256 hash of the magnet link created from the `WakuMessageArchive`.
+Community owner nodes MUST create a torrent file ("torrent") containing metadata to all message history archives. To create a torrent file, and later serve the message archive data in the BitTorrent network, community owner nodes MUST store the necessary data in dedicated files on the file system.
 
-The `magnet_uri` field MUST contain the magnet link created from the `WakuMessageArchive`.
+A torrent's source folder MUST contain the following two files:
+
+- `data` - Contains all protobuf encoded message history archives concatenated in ascending order
+- `index` - Contains the protobuf encoded message history archive index
+
+Community owner nodes SHOULD store these files in a dedicated folder that is identifiable via the community id.
+
+Community owner nodes MAY remove older torrent files that were generated for previous message history archives, after a new torrent was created.
```

Comment:
Shouldn't the removing of the previous (old) torrent file be automatic after the new torrent file is created?  Is there any reason not to do this?

Reply:
That's a good point. The new/latest torrent file will replace the current one, making this paragraph obsolete.